python/requirements_test.txt

```diff
@@ -1,6 +1,8 @@
 pytest==7.1.2
 pytest-randomly==3.8.0
-scipy==1.10.1
-tensorboard==2.13.0
+scipy==1.10.1; python_version < "3.12"
+scipy==1.11.4; python_version >= "3.12"
+tensorboard==2.13.0; python_version < "3.12"
+tensorboard==2.16.2; python_version >= "3.12"
```

Comment:
TensorBoard version must match the TensorFlow version. Supporting different tensorflow or tensorboard versions in a single release is not feasible.

Reply:
I see. tensorflow version comes from that other file, `ci_utils.sh`. Might not be a clean way to match these if we differentiate between versions like this

Reply:
The requirements files are the "source of truth" for versions. For other places (e.g. CI), we just need to ensure that the versions match.

PS: Different versions of scipy are OK since we only consume the Python API, unlike with tensorboard / tensorflow where there is a much deeper C++ / Python interaction.

Reply:
Well tensorflow is not in any requirements file, that I can see at least. But I can make it match. I'm just saying, may not be that clean

Reply:
Per my latest mainline comment, I am seeing now that it exists as a requirement in the Open3D-ML repo, so that will need to line up as well. But in the build scripts, the ML requirements files are only used some of the time